src/panels/config/devices/device-detail/ha-device-card-mqtt.ts

```diff
@@ -0,0 +1,51 @@
+import { DeviceRegistryEntry } from "../../../../data/device_registry";
+import { removeMQTTDeviceEntry } from "../../../../data/mqtt";
+import {
+  LitElement,
+  html,
+  customElement,
+  property,
+  TemplateResult,
+  CSSResult,
+  css,
+} from "lit-element";
+import { showConfirmationDialog } from "../../../../dialogs/generic/show-dialog-box";
+import { HomeAssistant } from "../../../../types";
+
+@customElement("ha-device-card-mqtt")
+export class HaDeviceCardMqtt extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public device!: DeviceRegistryEntry;
+
+  protected render(): TemplateResult {
+    return html`
+      <mwc-button class="warning" @click="${this._confirmDeleteEntry}">
+        ${this.hass.localize("ui.panel.config.devices.delete")}
+      </mwc-button>
+    `;
+  }
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+  }
+
+  private async _deleteEntry(): Promise<void> {
+    await removeMQTTDeviceEntry(this.hass!, this.device.id);
+  }
+
+  private _confirmDeleteEntry(): void {
+    showConfirmationDialog(this, {
+      text: this.hass.localize("ui.panel.config.devices.confirm_delete"),
+      confirm: () => this._deleteEntry(),
+    });
+  }
```

Comment:
`showConfirmationDialog` returns a promise that resolves to `true` if user confirmed. So you can do something like

```ts
const confirmed = await showConfirmationDialog(…)

if (!confirmed) { return; }

await removeMQTTDeviceEntry(this.hass!, this.device.id);
```